### ISSUE TITLE
Themes

### ISSUE THREAD

**Issue by amyjko** [wordplaydev/wordplay#108]
## What's the problem?

There are many different audiences that we may want to entice on the landing page, and having the page be responsive to seasons, locals, and sensory preferences is one way to be responsive to those audiences.

## What's the design idea?

Allow the application to have customizable themes that creators can choose from, affecting background, fonts, spacing, and more.

## Who benefits?

All creators who might want to customize the tone and appearance of the website.

## Design specification

Themes are an issue that can be determined by designers of how the wordplay website overall looks and feels towards the users. This may require a lot of changes to the CSS variables and below are a few environmental themes that allow the user to feel overall  positive and inclusive to a range of eye visibility challenges. Themes must be inclusive for a wide range of users and are not limited to the environmental themes I’m sharing with you below. This excludes the already current theme of Wordplay as well as dark mode. 

IMPORTANT NOTE: All images in leaf, cloud, and snowflake theme are made up of letters, to embody and capture the original theme of “wordplay” global languages.

### Environmental Theme 1: Cloud

This theme is to resemble a cloudy day, the colors are very vibrant and has good contrast with one another to allow different users to see and feel the theme as intended. This theme’s goal was to bring a bit of a “calm cloudy day feel” giving our users a positive feeling while using the website. 

Below is the image of the homepage.

Homepage*
![CloudWP-Home -Final](https://github.com/wordplaydev/wordplay/assets/146573912/6c8298f9-a97b-4bd8-9d8b-aecb7d53a91b)

Overall colors:
Hex values: light yellow: #FFF3C2
Hex Value Dark blue: #2B547E
Hex value: White: #FFFFFF
Hex Value: Gray :#969696
Hex Value: Black :#000000

Explanation: The clouds on the homepage are gray. However as the clouds move from side to side. Coming closer to the sun in “Wordplay” the clouds would begin to turn white. To allow users to feel a “brightness” from the change of color regardless of their own visible ability. Clouds are different sizes and similar to the letters in the homepage, they may vary paces (slightly) as well as go through each other. New clouds come in from the side where my gmail icon is showing (right) and disappear from the opposite side as it crosses the page (left).

As noted above, though not shown, the letters seen in the homepage float similarly as the clouds, though not as frequent. (2 or 3 every 6 clouds or so). These letters will have the same shadowing as the clouds passing the sun. Another thing to note is that all the clouds are using letters/glyphs from other languages. I will include my glyph references at the bottom of all 3 themes.

Why the colors: Through research, I learned how a lot of colors may contradict each other. For example blue-yellow and red-green colorblindness are visible to their opposite. But through research I learned that these Hex values  particularly the  blue, and yellow. Would still appear visible due to its vibrant contrast. Users with blue-yellow color blindness would still be able to see the blue and yellow, though not as vibrant, would still ultimately see the colors as intended. But mostly to capture its seasonal theme of “clouds”. Its vibrant colors also allow for different devices to be able to have that bright visibility for users.

Below are other references of how the theme should look across pages:

*learn page*
<img width="1752" alt="cloudWP-Learn-scene-final" src="https://github.com/wordplaydev/wordplay/assets/146573912/b9740f93-e667-4210-8519-d36f46e8ef0b">

*project/gallery*
<img width="1750" alt="cloudWP-Project-Gallery - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/a7705a4e-ce6e-4f36-81b4-c4825af45014">

*donate/setting*
<img width="1750" alt="cloudWP-donate-setting - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/3d06a8c4-1e11-4ffe-a54c-38fc6619a0ec">

Unlike the other 2 themes, I purposely allowed the Donation page to be written without being placed within a large white textbox. This is because I want this theme to still include its flowy cloud theme by having the white letters against the background to still resemble the sky. 
Each theme has its own unique touch, for this theme, This is its touch. 

### Environmental Theme 2: Leaf

This theme is to resemble fall, the colors are very vibrant to allow different users to see and feel the theme as intended. This theme’s goal was to bring a bit of “fall feels” intending to give users a resting, and calm emotion upon use. I wanted to avoid the interpretation of thanksgiving and halloween in this theme to account for cultural differences. Since everyone experiences fall, I wanted to stick to making it just that. A fall/leaf theme.

Overall goal: We want our users to have a positive feeling while using the website. 
Below is the image of the homepage and Learn.

*homepage/learn*
![FallWP-Home -learn- final](https://github.com/wordplaydev/wordplay/assets/146573912/823efc05-207b-416e-9244-4ef0dcd8c8d4)

Overall colors:
Hex values:#F6BC31: orange gold
Hex values:#36563C: dark green
Hex values:#7D5AA3: Purple
Hex values: #A83F29: red
Hex values:#3A0603: brown/dark red

Explanation:
This is a fall theme, The leaves float randomly from the top and onto the bottom, disappearing. The leaf has a wide range of colors as fall is a very colorful season! Similar to the cloud home page the letters will fall like the leafs, however other than the homepage, it would not be as frequent. (maybe about 1 letter every 5 leaves or so users don't get distracted). The unique part about this theme’s design is also the tree. The tree sits onto the toolbar When users scrolls down, The tree would follow the toolbar. And just like how the clouds was made up of letters, so dies this tree and its leaves! 

Why colors:
Similar to the cloud theme, The orange color for the fall theme is very vibrant and will still be visible to users with red-green color blindness. Though not as vibrant. The goal was to still allow for users to see and feel the theme as intended. This theme would be very vibrant for those with blue-yellow color blindness and be fairly visible to those with red-green. While the cloud theme would work similarly and in the opposite. The leaves are a wide range of different, and vibrant colors to allow all users to see the fall theme as colorful as I intended it to look. 
The leaves in this design are a bit mixed in color to show the changed color of the leaf during this beautiful season, to be clear some of the letters that make up the leaf are in different color to show this.

For the Learn Page, The leaves will float randomly on the top from the top to the bottom in different sizes, you can also see that on the bottom there is a pile of fallen leaves, when animating the leaf’s motions, the pile of leaves don’t move, so to not distract users from learning as well as to represent our favorite, piles of leaves in fall.

Other parts of the theme:
*learn/projects*
<img width="1751" alt="FallWP-Learn-Projects - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/bd272df0-f49a-4993-8d4f-5ee86fa5c789">

As you can see from the projects and gallery theme: 
As you can see from the projects and gallery theme: 
I have changed the add box to make it bigger and easier for the users to see and read its content. Another thing to note, there is something else sitting on the toolbar of this page. I wanted to add a bit of a story in this theme, as the leaves fall. We would need to pick it away. So a leaf rake sits on the toolbar of this gallery page, and alongside it is a pile of leaves.
(The leaf falls in a windy motion and disappears, once again the pile of leaves would not move, however occasionally some leaves would join the pile).

Here are how other references/parts of the theme look like:
*gallery*
<img width="1750" alt="FallWP-Galleries - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/3629d414-2d3a-4e97-b60f-4754be0451ca">

*donate/setting*
<img width="1750" alt="FallWP-Donate-Settings - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/a7c01e8a-46b9-4663-81ca-ebf6ce221fd6">

### Environmental Theme 3: Snowflake

This theme is to resemble winter, the colors are very vibrant to allow different users to see and feel the theme as intended. This theme’s goal was to bring a bit of “snowy winter” . I wanted to give users a more ”Icy feel” with this theme so for the home page I used a icy blue color for the toolbar where the snowman sits on. The Toolbar on the homepage is the only one like this, it goes back to its normal white color for other pages. 

Overall goal: We want our users to have a positive feeling while using the website. 
Below is the image of the homepage.

*homepage/learn*
<img width="1750" alt="winterWP-home-learn - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/9a33be6d-122c-4090-8b02-972a6e1ad431">

Overall Colors:
Hex values: Icy blue; #DCEFEF
Hex values: Gray; #434D60
Hex values: brown;#CC5500

Explanation:
In the homepage, you can take notice of the snowman sitting on the left of the screen. On top of the toolbar. Similar to the tree on the homepage the snowman is made up of random letters of different languages. I kept the “f” mascot glyph as its face, and I didn’t include buttons, a hat, and a scarf because I wanted to avoid any cultural differences attached with a snowman. As for the snowflakes they will fall from above and onto the ground. The toolbar is icy blue to make it seem like the snow is piling up. The snowflakes fall in various sizes and do not go through each other. The letters will fall less frequently, 1 in every 4 snowflakes. Letters should be white in color. 

Why colors: 
The icy blue and white may seem too similar of colors depending on one's visual ability but its overall meaning should still come across regardless of that. I also used a brownish vibrant color in some areas such as buttons/signifiers of the website to give a bit of a woody/wintery feeling.

The background of this theme is a dark grayish color, this is to represent the winter gray color but also to have more inclusivity with color blindness, I didn’t go for a lighter gray because a vibrant contrast is needed for visibility and I didn’t want it to be too similar to the original (light) theme in regards with its color scheme of white and gray being its majority.

Other parts of the theme:
*project/gallery*
<img width="1751" alt="winterWP-project-gallery - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/0028a22a-f3f0-4bc4-a24b-910f5988e250">

Snowflakes fall, downward. Size varies. 

*donate/setting*
<img width="1750" alt="WinterWP-Donate-Setting - final" src="https://github.com/wordplaydev/wordplay/assets/146573912/60d60c6d-2e3d-4d6e-8b32-795ddb929dae">

These are 3 themes outside of 2 already existing themes (light, and dark) These environmental themes do not have a light or dark  mode attached to them. In settings next to light or dark mode exists a cloud icon, a leaf icon, and a snowflake icon which can be seen in the mockups above.

As shown in the mockups, certain colors act as signifiers, background, ect.. the references of other pages are here to show how colors should be typically applied.

As for animations of each theme, I ask developers to allow motion with the images (leaf, snowflake, clouds) as it would naturally in the environment around us (wind like motions for fallen leaves, slowly floating clouds, ect.)

Sample glyphs:
Д |  አ ማ ር ኛ |   ا ل ع رب ي ة |  অ স মী য়া |  ə |  ш ҡ |  Б | б | л | ъ | বাং  লা | བོ ད་ ཡི ག | č | š | ε λ | ف ا رس ى | ø | ç | हिं दी | Հ ա յ ե ր ե ն  |   ꆈ ꌠ ꁱ ꂷ | ᐃ ᓄ ᒃ ᑎ ᑐ ᑦ (ᑲ ᓇ ᑕ) | 日 本 語 | ქ ა რ თ უ ლ ი | қ ш | ខ្  មែ រ | ಕ ನ್ ನ ಡ | 한 국 어 /韓 國 語 | कों क णी | г | ë | ລ າ ວ | മ ല യാ ളം | ᠮ ᠤ ᠨ ᠭ ᠭ ᠤ ᠯ ᠬ ᠡ ᠯ ᠡ | म रा ठी | ने पा ली (ने पा ल) | ਪੰ ਜਾ ਬੀ | پ ښ ت و | ê | â | ă | й | सं स्कृ त | සිං හ ල | č |  ܣ ܘ ܪ ܝ ܝ ܐ | த மி ழ் | తె లు గు | ҷ | ไ ท ย | ü | ї | Ў | 㗂 越  |

End.

**Comment by amyjko** [comment]
@RuweydaAbdi, I believe someone else was working on this with you? Please make sure everyone is assigned to this issue so that we know whose working on it.

**Comment by RuweydaAbdi** [comment]
Here is our draft proposal which includes:

- Problem/Goal we want to solve
- Types of themes we have created
- Animations we want included 
- Certain RGB colors
- And Overall Research

Summary: 
We are creating environmental Wordplay themes that are aiming to be inclusive towards those with different types of color blindness such as red-green and blue-yellow color blindness. Our goal was to choose very vibrant colors that allows Individuals with this different types of color blindness to be able to and experience similar colors as those without this disability. Throughout the quarter we aimed to figure out what types of themes we wanted to be included into the Wordplay website. We came to the conclusion that rather than changing the overall layout of wordplay, we want to include different and inclusive environmental themes that users would be able to choose by going into the settings and clicking on different icons that resemble each chosen theme. EX/ cloud theme → cloud icon. 
The current type of themes we have created (which would all be finalized a bit further over the break) are as listed below:

- Cloud theme
- Fall theme
- Star theme
- Winter theme
- Plant theme 
- High contrast theme

We recognize that there are no current limits on themes for the site. Keeping in mind that we also wish to keep the light and dark mode options that are currently and already in place on the site. 

Current problems with our themes: It takes away the overall theme of the site which is words. 
Solution goals: We wish to find a way to include words along with our environmental themes.

Other problems: We recently discovered that different laptops showcase our chosen RGBs differently. 
Solution: We found that cheaper/ low quality laptops generally are less vibrant in color than higher quality laptops such as macbooks. Cheaper displays only cover 50-60% of sRGB rather than 90-100% covered by higher-quality panels. Through our research we found that there is no specific named laptop that we could test this on when it comes to choosing RGBs. However we are aware of the laptops with specified resolutions. For example, laptops with 1366x768 or 1920x1080 resolutions tend to have lower quality panels compared to higher resolutions like 2560x1440 or 3840x2160. In other words, higher-end laptops tend to have better displays. 

In order to find vibrant RGB colors that allow for inclusivity of those with certain types of color blindness. We would need to test out all off our RGB’s on laptops that are relatively of cheaper quality such as 

1. Dell Inspiron and Dell Latitude laptops which have a resolution of 366x768 or 1920x1080 displays. — of lower display quality,
           * Laptops in SPS tend to be Dell laptops. 

          * Wordplay's target audience is middle- High school, those with color blindness may be students who use the site on dell, so experimenting with RGB’s on this laptop may give us great results.
          
2. Asus Vivobook — The laptop I use is of medium quality with an average resolution of 1920x1080 display, not considered low. 
        * Most laptops are of 1080 display
        
       * Very dull RGB compared to Katherine’s Macbook
       
3. Macbook pro —- All current MacBook Pros have either 2560x1600 or higher resolution displays. Meaning of higher quality and shows the most vibrant colors than all listed.
       
      * The laptop we want to avoid testing RGB’s on

Why is having vibrant RGBs important for those with certain color blindness?

1. Seeing vibrant colors allows those with certain color blindness to be able to see clearer differences of the chosen colors with their remaining color vision.
2. Having chosen certain RGB that are not as vibrant in other’s screens may cause the design of our theme to not come off fully as intended. Which conflicts with our goal of having those with color blindness to be able to see the colors we have chosen as intended with those without this disability.
3. Most colorblindness are still able to see certain shades of the same color. Having vibrant RGBs allow for these individuals to be able to take quick notice of those different shades. 


Animations:
Each theme we have designed, along with different RGBs may also include the different types of animations that correspond with our chosen themes. Each Theme to be posted would include 

- RGBs used in each page 
- The types no animations used
- Different images that are included into the design. 
- And our overall intention of the design.

We have done some research on CSS variables that allow for our animations ideas to essentially come to life. 
For example; Cloud theme animations is to have clouds move from side to side on the top of the page.
This CSS should allow for the icon to move back and forth. 
“ --icon-x-pos”.
Our goal is to be as specific as we can about our designs and the type of images and animations we want for the site.

Overall research: 

Design Decisions for Color Calibration: Designers have to make decisions about color calibration settings for displays. If certain laptops consistently show dimmer colors, designers may need to calibrate their designs to look acceptable on a broader range of devices, including those with lower-quality displays.

Research and User Testing: To address this issue, designers must conduct research on different laptop models, understand their display specifications, and conduct user testing to ensure that designs appear as intended across various devices. This may involve creating multiple design versions to account for variations in display quality.

NOTE:
Themes will be edited/Finalized and posted here prior to the first day of Winter quarter.

**Comment by amyjko** [comment]
This draft is a good start! You've done a good job brainstorming the details you need to specify in the final proposal.

A few comments, as you prepare your next iteration:

* Make sure to post the draft in the main body of the issue, not as a comment. (This is an old issue, and so the body of the issue should reflect the current feature request template. Click on "new issue" and choose "enhancement" to see the template).
* Ensure your design proposal includes all of your mockups. Developers will need all of them to implement your vision.
* For each theme, there should be exact details on its appearance and behavior, including exact colors for the light and dark modes. There should also be exact specifications for what animations you're expecting, so that developers do not have to design the animations. You might capture these in a GIF.
* I'd recommend focusing this issue on the smallest set of themes you believe would meet the goals; more isn't necessarily better, since each one is unique enough that it would take a lot of effort to build. What's the smallest set that you think achieves the design goals?
* I recommend organizing your design specification around the individual themes, and the theme chooser, rather than elements of each theme. Developers will have to build each of these components separately, so organizing the information around the design, rather than facets of each design, will make it easier to see how each design comes together as a whole.

In general, remember that developers should not have to design _anything_ to build these. Every detail of appearance and interaction design is up to you to specify. Expect many iterations with me until we have all of those details down.

**Comment by RuweydaAbdi** [comment]
Thank you, I'll be sure to fix up the draft soon! :)

**Comment by RuweydaAbdi** [comment]
@amyjko

**Comment by amyjko** [comment]
@RuweydaAbdi Remember when you tag me to say what you're asking for. I manage a thousand emails a week and hundreds of GitHub updates, so I need as much context as possible to help me remember what you're asking for :) (I assume here you're asking for design proposal feedback, but it'd be helpful to know what changed, if there are any issues you haven't resolved yet, etc).

**Comment by RuweydaAbdi** [comment]
@amyjko Oh ok so sorry, Yes I would like some design feedback on the three themes I submitted. Animation is a big part of what has changed. I would like to know if the animation designs (homepage especially) has any issues as well as whether the dev team are able to add these animations for each theme separately. I'm also curious how it would look like once implemented. Would it be too distracting for the users who for example have issues with focusing. (Something I struggle a lot with as well). I also asked for the type of speed I think would be fine for each theme. However i did not give any numbers just a general "feel" of how fast it should be. I would like to see and test it in a way with the dev team.

**Comment by amyjko** [comment]
Thanks @RuweydaAbdi! I've reviewed the design, and like the overall direction. Here are several questions that have to be resolved before we can build anything:

* The image assets you've included via PDF can't be easily extracted and reused. They should be included as individual image assets, not embedded in a PDF.
* The image assets you've included need to be a resizeable image format, like SVG, since the application must be viewable on many different sizes of devices.
* I'm a bit concerned about the use of images for the themes. Accessibility is a central goal of the website, and so we generally try to avoid any images, as they would all require localized image descriptions in all of our supported languages in order to be accessible to people who are blind or low vision and rely on screen readers. If we are going to use images for these, then we need image descriptions for every asset, in every language we support, and precise details on which images should be screen readable, and which should be invisible to people who use screen readers. An alternative to images is to use text glyphs or emojis. For example, you could design a tree out of letters, use a leaf emoji, etc. That would be more consistent with the platforms focus on text and typography.
* Overall, the colors are nice, but I'm less sure about the yellow text on the blue background. Aesthetically, it's not great, and I'm unsure about its contrast levels for readability.
* Across all of the themes, its unclear where the color choices apply. Is it the background and font color of _every_ page in the application, or just the specific ones you've specified? If the former, that should be explicit, and if the latter, why only those pages?
* How do all of these themes interact with dark mode? The current theme has two sets of colors for light and dark mode, but I didn't see any mention of how dark mode would be handled for any of these. Are you viewing dark mode as just another theme, or is the expectation that all of the themes also have a dark mode? If the latter, then colors should be specified for dark mode for each. The settings mockup you should seems to append them onto the lis of light, dark, and automatic modes, but the point of dark mode is to support creators with light sensitivity, and not having a dark mode for each theme would exclude them from using them.
* In the mockups, header colors aren't consistent. Sometimes they're black, sometimes they're the theme color. Is that intentional, or just a flaw in the mockups?
* Animations are definitely fine to include, but they should be more precisely specified. Exactly what motion do you want developers to implement? Can you make a mockup, so it's unambiguous? (You don't want developers to design the animations, since they're not thinking about it from the perspective of the tones and feelings you're trying to convey).

Let me know if you have any questions about these!

**Comment by amyjko** [comment]
It's the end of Winter! Please provide an update on this issue, including:

* Whether you plan to continue work on it
* If you do not, 1) provide a detailed update of its status, 2) add any unfinished work, 3) list any any unmerged branches, and 4) unassign yourself, so someone else can take over the task. Please leave the issue in better shape than you found it, helping future contributors to start from where you left off.

If you do plan to continue work on it, carry on. Otherwise, thank you for your contributions!

**Comment by RuweydaAbdi** [comment]
I am currently finishing touches and will send in my proposal in soon with all the corrections.

**Comment by amyjko** [comment]
Wonderful! Thank you @RuweydaAbdi, I'm excited to see the next iteration! Maybe we'll find someone to build in it Spring :)

**Comment by RuweydaAbdi** [comment]
@amyjko

**Comment by amyjko** [comment]
Hi @RuweydaAbdi. You tagged me? Is this a request to review? (When you tag someone, make sure your request is clear, especially when you tag someone that who is tagged by many people).

**Comment by RuweydaAbdi** [comment]
Sorry, I realized I forgot to tag you earlier so i did it quickly, I was so tired last night I didn't realize I only tagged you.  
I updated my designs of the three themes cloud, leaf, and snowflake I'd like your feedback on it. I also made the images into a PNG so it could be extracted easier. 
I wanted to make all the images out of letters to play with words 
But would emojis be better and simpler for developers?

**Comment by amyjko** [comment]
The designs are so cool! They really look incredible. Don't worry about what's easier for developers; design is all about what we should make and why, developers get to figure out how.

One thing that will be necessary for developers, however is knowing what specific glyphs you used in all of the designs, and the positioning of those glyphs, so they can recreate the designs from the sketches you've provided. One way to provide this would be to attach a source file, so they could extract the glyphs and positions from those sources into a format that could be used to render everything.

I think the animation specifications are a bit open-ended — developers could animate things in many different ways — but it's hard to specify those exactly, and there are many different ways to do it well, so I'd recommend whoever develops this just work with designers to ensure the animations capture the sprit of the design.

One last thing: I mentioned the need for descriptions of the content for blind and low vision users relying on screen readers. Can you write English descriptions of each of the objects you designed, so we can translate those into other languages, and figure out how to make them screen readable? (By default, a screen reader would read out individual glyphs, which isn't what we want).

**Comment by RuweydaAbdi** [comment]
Thank you! @amyjko 
I'll try to be a bit more specific with the animations to give developers an idea of what I'm looking for and I'd be happy to work with them with the design's implementation. I'm a bit confused about a source file? does this mean a google document where i say which glyph goes where in each image? Also for the screen readers, does this mean in every image I should put like a caption text under each image for example; under a leaf, I write "leaf".

**Comment by amyjko** [comment]
By "source file", I meant the original document that you used to create the images (e.g., Illustrator).

For screen readers, yes, I meant language to describe the typographic arrangements you designed; sighted people can infer what they are with visual perception, but unsighted people cannot, so they need a textual description of what the object is.

**Comment by RuweydaAbdi** [comment]
Oh ok, I'll add in the texts and for the source file, I used paint. Not a lot of people use windows ( do apple products have the paint app?) Would that cause a problem for developers opening it?

**Comment by amyjko** [comment]
Thanks!

Re: Paint, if all you have is an image file from Paint, then no need to share that, since there's no additional information it has beyond pixels. (There are plenty of macOS apps for editing image files).

The hard part for building this will be recreating your designs; it will be up to developers to come up with a glyph by glyph recreation of exact positioning to be able to remake the designs you made. They'll have to use your images for reference, and that will take a long time. The only way to really make that easier would be to create that for them: for each glyph, a relative `x`, `y` position. It's either you or them :)

**Comment by RuweydaAbdi** [comment]
oh yikes, ok. I'll see what I can do to make it easier by adding coordinates. I had affinity design for a while before my free trial ended. If I click "open with" on my paint file I would have had the option to see it in affinity. I remember there being x,y coordinates when you tap at a specific part of an image. I can try to get another trial and see if that makes things easier.

**Comment by amyjko** [comment]
@alexhuynh0 Don't forget to say why. Capturing rationale is very important, so we know why we made a decision, and not others.

**Comment by alexhuynh0** [comment]
I am not continuing work, and I am including all my team's (Ruweyda Aldi (designer), Leo Tian, George Lee (mentor), Nathan Dang, Matthew shell) work materials in the issue so that others can resume where we left off. The work material consists of messages we left in a discord channel "issue-108" that will go away after its been inactive.

![C4hGBEvy](https://github.com/wordplaydev/wordplay/assets/131318750/9f44689b-b98a-42be-9bf8-b744b3f6c8ec)

![BVjRE07p](https://github.com/wordplaydev/wordplay/assets/131318750/8955b9f3-ad48-4866-bbb6-f2bea7f067e3)

![YGWX4j8F](https://github.com/wordplaydev/wordplay/assets/131318750/094dce55-4c33-4b7c-89e7-998d8f607735)

![y7gBVXC7](https://github.com/wordplaydev/wordplay/assets/131318750/e3a838f3-1729-49e1-b6c0-c1bcb0141f7f)

![8JSQChnd](https://github.com/wordplaydev/wordplay/assets/131318750/4ff5b59e-a30d-4776-85e5-5a90ffd81c80)

![pUqqEXa9](https://github.com/wordplaydev/wordplay/assets/131318750/72458718-44e7-4629-b399-3a3c022dd307)

![UzhZSqLs](https://github.com/wordplaydev/wordplay/assets/131318750/2c1f067e-0717-44c6-b8fb-b9d49e38cd67)

![6IZHIKe1](https://github.com/wordplaydev/wordplay/assets/131318750/7dca6687-d30a-41c8-91c8-9e55e6756c3a)

![xWtTuA0b](https://github.com/wordplaydev/wordplay/assets/131318750/0814f5dc-a6d6-48a3-9782-3bbb46e18116)

![TtoEGcsz](https://github.com/wordplaydev/wordplay/assets/131318750/1cf43023-7106-4c39-9444-bdd0f020501e)

![image](https://github.com/wordplaydev/wordplay/assets/131318750/901b50a4-a960-4239-a7ea-30ade0a178c7)

**Comment by leo3tian** [comment]
Thank you for the overview @alexhuynh0! I will be carrying on work with this issue in fall quarter. One additional thing I would add is that we can draw from a lot of the existing code in `Background.svelte` that generates glyphs that fly across the homescreen, and use the same system but with different colors, glyphs, and movement.

**Comment by shellmatthewk** [comment]
I don't have much more to add since I spent quite a bit of time reviewing svelte. I am unsure if I will be continuing work on this issue in autumn, but one thing I would like to suggest for anyone else picking this issue up is to review the svelte docs early on in the quarter. Some of this issue can be found here: https://learn.svelte.dev/tutorial/component-styles, and this other documentation that @alexhuynh0 created. 
[SvelteWebpageTutorial.docx](https://github.com/wordplaydev/wordplay/files/15491828/SvelteWebpageTutorial.docx)
